sync/_databricks.py

```diff
@@ -495,6 +499,82 @@ def record_run(
         )
 
 
+def apply_prediction(
+    job_id: str, project_id: str, prediction_id: str = None, preference: str = None
+):
+    """Updates jobs with prediction configuration
+
+    :param job_id: ID of job to apply prediction to
+    :type job_id: str
+    :param project_id: Sync project ID
+    :type project_id: str
+    :param prediction_id: Sync prediction ID, defaults to latest in project
+    :type prediction_id: str, optional
+    :param preference: Prediction preference, defaults to "recommended" then "economy"
+    :type preference: str, optional
+    :return: ID of applied prediction
+    :rtype: Response[str]
+    """
+    if prediction_id:
+        prediction_response = get_prediction(prediction_id, preference)
+    else:
+        predictions_response = get_predictions(project_id=project_id)
+        if predictions_response.error:
+            return predictions_response
+        prediction_id = predictions_response.result[0]["prediction_id"]
+        prediction_response = get_prediction(prediction_id, preference)
+
+    if prediction_response.error:
+        return prediction_response
+
+    prediction = prediction_response.result
+
+    databricks_client = get_default_client()
+    job = databricks_client.get_job(job_id)
+
+    job_clusters = _get_project_job_clusters(job)
+    project_cluster = job_clusters.get(project_id)
+
+    if not project_cluster:
+        if len(job_clusters) == 1 and None in job_clusters:
+            project_cluster = job_clusters[None]
+        else:
+            return Response(
+                error=DatabricksError(
+                    message=f"Unable to locate cluster in job {job_id} for project {project_id}"
+                )
+            )
+    project_cluster_path, _ = project_cluster
+
+    if preference:
+        prediction_cluster = prediction["solutions"][preference]["configuration"]
+    else:
+        prediction_cluster = prediction["solutions"].get(
+            "recommended", prediction["solutions"]["economy"]
+        )["configuration"]
+
+    if "cluster_name" in prediction_cluster:
+        del prediction_cluster["cluster_name"]
+
+    if project_cluster_path[0] == "job_clusters":
+        new_settings = {
+            "job_clusters": [
+                {"job_cluster_key": project_cluster_path[1], "new_cluster": prediction_cluster}
+            ]
+        }
+    else:
+        new_settings = {
+            "tasks": [{"task_key": project_cluster_path[1], "new_cluster": prediction_cluster}]
+        }
+
+    response = databricks_client.update_job(job_id, new_settings)
```

Comment:
My only comment on this is based on the databricks api documentation:

```Changes to the field JobSettings.timeout_seconds are applied to active runs. Changes to other fields are applied to future runs only.```


I don't believe we ever touch timeout_seconds but do we want to be extra cautious and delete it from `new_settings`? its a little bit silly but figured id call it out.

Reply:
Yeah I think that's a good idea here especially since the backend merges prior configs with predictions in the prediction API result - it could be set here. Less of a concern for recommendations but probably an easy fix there too.

Reply:
Actually, we may be good here. The documentation only mentions `JobSettings.timeout_seconds` while we only touch clusters within `JobSettings`.
https://docs.databricks.com/api/workspace/jobs/update